flink-runtime/src/main/java/org/apache/flink/streaming/api/datastream/DataStream.java

```diff
@@ -864,7 +864,11 @@ protected DataStream<T> setConnectionType(StreamPartitioner<T> partitioner) {
      *
      * @param sinkFunction The object containing the sink's invoke function.
      * @return The closed DataStream.
+     * @deprecated This method relies on the {@link SinkFunction} API, which is due to be removed.
```

Comment:
nit: maybe change which is due to be removed. -> which is deprecated.
The phrase  "due to be removed" is ambiguous, are we saying v2 will have this as deprecated. Assume this would then be removed in Flink v3.

Reply:
Yes. Because the `SinkFunction` will be removed, this function depends it.

ApiAnnotationRules lead to the error https://github.com/apache/flink/pull/26297/files#r1996509218
It means we should add `@Deprecated`